Automattic/wp-calypso
# Components: Make compact toggle the canonical one

As per p9Jlb4-1Q9-p2 we're working to unify our toggles and get them to use `@wordpress/components`. We already removed all instances of the large toggle in #46653, so here we're making the compact one be the canonical one.

Furthermore, we're fixing some ESLint rules when passing by.

#### Changes proposed in this Pull Request

* Make the compact toggle the canonical one, replacing the large one by default.
* Update all instances of the compact toggle to now use the canonical toggle one.
* Remove the compact toggle component that is no longer necessary.
* Remove unnecessary component tests, related to the compact toggle functionality.
* Fix some import path ESLint warnings.

#### Testing instructions

This PR should introduce no visual or functional differences.

Go through the following pages with toggles and verify the toggles look and work the same way they do on production:
* `/devdocs/blocks/post-share` - all toggles in the example
* `/following/manage` - the toggles in the Settings popover of a site you're following
* `/settings/taxonomies/category/:site` - click on "Add New Category" and observe the "Top level category" toggle
* `/devdocs/design/form-fields` - the toggle example
* `/devdocs/design/g-suite` - toggles at the bottom of the example
* `/devdocs/playground` - try inserting a `<FormToggle />` in the DevDocs playground and you should see a working compact toggle.
* classic Calypso editor - the "Allow people to buy more than one item at a time." toggle when creating a simple payments product
* classic Calypso editor - the toggles in the sidebar, whether to make the post sticky, the "Request review for post" toggle when the post is pending, and the "Toggle to set as top level" one.
* `/store/product/:siteSlug/:productId`, where `:siteSlug` is the site slug of a site with WooCommerce, and `:productId` is the ID of a product, observe the "product with variations..." and "Promote this product across the store" toggles.
* `/store/product/:siteSlug/:productId`, where `:siteSlug` is the site slug of a site with WooCommerce, and `:productId` is the ID of a product with variations; click on the name of the variation and in the modal, observe the "Visible" toggle.
* `/store/settings/taxes/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services active, and a valid USA address selected as the store address, observe the "Tax calculations enabled" toggle.
* `/store/settings/shipping/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services active, observe the toggle at the right of the "Shipping Labels" card heading.
* `/store/settings/email/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce, WooCommerce Services, and Mailchimp for WooCommerce active, observe all the toggles on the page.
* `/store/settings/payments/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services, observe the toggles next to each of the available payment methods.
* `/store/settings/shipping/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services, observe the toggles in the modal when editing a shipping method, in the list of shipping methods, and in the shipping labels section.
* `/extensions/wp-super-cache/:siteSlug` where `:siteSlug` is a site slug of a Jetpack site with WP Super Cache installed. Go through all tabs and look at all toggles.
* `/me/privacy` - the toggle at the bottom of the "Usage Information" card.
* `/me/purchases` - select a purchase, and observe the toggle in the Subscription Renewal section (the rest of the auto-renew toggle instances in the domain management section were already compact ones).
* `/domains/manage/:siteSlug/contacts-privacy/:domain` where `:siteSlug` your site and `:domain` is a domain on that site, observe the "Privacy Protection" toggle
* `/domains/manage/:siteSlug/dns/:domain` where `:siteSlug` is your site and `:domain` is a domain on that site, observe the domain connect toggle. It's possible it can be disabled if the domain is managed by WP.com currently
* `/domains/manage/:siteSlug/name-servers/:domain` where `:siteSlug` your site and `:domain` is a domain on that site, observe the "Use WordPress.com Name Servers" toggle
* `/earn/ads-settings/:siteSlug` where `:siteSlug` is the slug of a site that has WordAds enabled. Observe the multiple toggles on this page
* `/earn/payments-plans/:site` where `:site` is the slug of a simple WP.com site with a business plan. You will have to authorize Stripe first before accessing that screen. When there, click on the "Add a new payment plan" button and observe the two toggles in the modal.
* `/plugins/manage/:siteSlug` where `:siteSlug` is the slug of a Jetpack site with plugins. Observe the toggles on each plugin.
* `/settings/discussion/:siteSlug` - where `:siteSlug` is the slug of a site; go through all the settings sections for both a WP.com site and Jetpack site (ideally both with high tier paid plans) and observe the toggles.
* `/marketing/traffic/:siteSlug` - where `:siteSlug` is the slug of a site; there are multiple toggles on that page in (especially if you're on a high-tier paid plan). 

Finally, verify all tests pass.

#### Notes

ESLint warnings are expected. I consciously ignored most of them to avoid making the PR too huge to review.
